2.evaluate_data/nbconverted/predict_sc_probabilities.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Predict Single Cell Probabilities
+# In this part of the analysis, the single cell probabilities are predicted and stored in a parquet file.
+# The index of this file is used to reference the single cell index in the corresponding parquet files, where the "Metadata_plate" column refers to the plate name corresponding to the reference file.
+
+# ### Import Libraries
+
+# In[1]:
+
+
+import gzip
+import io
+import pathlib
+
+import pandas as pd
+import requests
+from joblib import load
+
+
+# ## Find the path of the git directory
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Load data code
+
+# In[3]:
+
+
+def load_joblib_from_url(url):
+    """
+    Retirieve joblib or gzip csv file from url
+
+    Parameters
+    ----------
+    url : url
+        The raw url of the file
+
+    Returns
+    -------
+    obj : any Python object
+    """
+
+    response = requests.get(url)
+    content = response.content
+    file_object = io.BytesIO(content)
```

Comment:
```suggestion
    file_object = io.BytesIO(response.content)
```
avoid using single-used varaibles in order to save more memory.

Reply:
Changed!